internal/pd/get-shifts.go

```diff
@@ -0,0 +1,115 @@
+// Copyright © 2020 The Homeport Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package pd
+
+import (
+	"strconv"
+)
+
+//Requirement: Shifts must be sorted in .pd.yml file
+
+//Shift specifies time range and name of a shift
+type Shift struct {
+	Start Time
+	End   Time
+	Name  string
+}
+
+//Time stores hour and minute of a specific moment
+type Time struct {
+	Hour   int
+	Minute int
+}
+
+//GetNextShift returns information about the next shift and time until it starts
+func GetNextShift(shifts []Shift, shiftPos int, currentTime Time) (Shift, Time, error) {
+
+	var nextShift Shift
+	if shiftPos+1 < len(shifts) {
+		nextShift = shifts[shiftPos+1]
+	} else {
+		nextShift = shifts[0]
+	}
+
+	timeUntilNextShift := Time{0, 0}
+
+	timeUntilNextShift.Hour = nextShift.Start.Hour - currentTime.Hour
+	if currentTime.Minute > nextShift.Start.Minute {
+		timeUntilNextShift.Hour--
+		timeUntilNextShift.Minute = 60 - currentTime.Minute + nextShift.Start.Minute
+	} else {
+		timeUntilNextShift.Minute = nextShift.Start.Minute - currentTime.Minute
+	}
+	if currentTime.Hour > nextShift.Start.Hour {
+		timeUntilNextShift.Hour += 24
+	}
+
+	return nextShift, timeUntilNextShift, nil
+}
+
+//GetCurrentShift returns all shifts in a slice and the position of the current shift
+func GetCurrentShift(currentTime Time) ([]Shift, int, error) {
+
+	shifts, err := LoadShifts()
+	if err != nil {
+		return []Shift{}, 0, err
```

Comment:
Two suggestions here since we are in an error case:
- It does not really matter here, but in other projects or scenarios, creating an empty slice of a custom type costs more time for the computer rather than just returning a `nil` instead.
- When returning an index from a list, there is no rule but a very strong preference in programming to use `-1` as an indicator that there was no hit. Mainly, because 0 is technically a valid index. So in error cases, consider using an invalid index to highlight the fact it is an error case.
```suggestion
		return nil, -1, err
```